src/components/Router/StepComponentMap.ts

```diff
@@ -267,7 +267,9 @@ const buildDocumentComponents = (
   )
   const showCountrySelection =
     showCountrySelectionForSinglePreselectedDocument ||
-    (!hasOnePreselectedDocument && !supportedCountry && countryCode !== null)
+    (!hasOnePreselectedDocument &&
+      !supportedCountry &&
+      (countryCode !== null || configForDocumentType === true))
```

Comment:
I believe this could be broken into smaller & more verbose checks :thinking:

Reply:
Agree. I've taken another stab at it, but if you have any ideas for further improvements, feel free to suggest.